src/utilities/mapSchemaConfig.ts

```diff
@@ -0,0 +1,464 @@
+    mappedConfig = mapper == null ? mappedConfig : mapper(mappedConfig);
+    return new GraphQLUnionType(mappedConfig);
+  }
+
+  function mapEnumType(type: GraphQLEnumType): GraphQLEnumType {
+    const config = type.toConfig();
+    let mappedConfig: Maybe<GraphQLEnumTypeMappedConfig> = {
+      ...config,
+      values: () => {
+        const newEnumValues = Object.create(null);
+        for (const [valueName, value] of Object.entries(config.values)) {
+          const mappedValue = mapEnumValue(value, valueName, type.name);
+          newEnumValues[valueName] = mappedValue;
+        }
+        return newEnumValues;
+      },
+    };
+    const mapper = configMapperMap[SchemaElementKind.ENUM];
+    mappedConfig = mapper == null ? mappedConfig : mapper(mappedConfig);
+    return new GraphQLEnumType(mappedConfig);
+  }
+
+  function mapEnumValue(
+    valueConfig: GraphQLEnumValueConfig,
+    valueName: string,
+    enumName: string,
+  ): GraphQLEnumValueConfig {
+    const mappedConfig = { ...valueConfig };
+    const mapper = configMapperMap[SchemaElementKind.ENUM_VALUE];
+    return mapper == null
+      ? mappedConfig
+      : mapper(mappedConfig, valueName, enumName);
+  }
+
+  function mapInputObjectType(
+    type: GraphQLInputObjectType,
+  ): GraphQLInputObjectType {
+    const config = type.toConfig();
+    let mappedConfig: Maybe<GraphQLInputObjectTypeMappedConfig> = {
+      ...config,
+      fields: () => {
+        const newInputFieldMap = Object.create(null);
+        for (const [fieldName, field] of Object.entries(config.fields)) {
+          const mappedField = mapInputField(field, fieldName, type.name);
+          newInputFieldMap[fieldName] = mappedField;
+        }
+        return newInputFieldMap;
+      },
+    };
+    const mapper = configMapperMap[SchemaElementKind.INPUT_OBJECT];
+    mappedConfig = mapper == null ? mappedConfig : mapper(mappedConfig);
+    return new GraphQLInputObjectType(mappedConfig);
+  }
+
+  function mapInputField(
+    inputFieldConfig: GraphQLInputFieldConfig,
+    inputFieldName: string,
+    inputObjectTypeName: string,
+  ): GraphQLInputFieldConfig {
+    const mappedConfig = {
+      ...inputFieldConfig,
+      type: getType(inputFieldConfig.type),
+    };
+    const mapper = configMapperMap[SchemaElementKind.INPUT_FIELD];
+    return mapper == null
+      ? mappedConfig
+      : mapper(mappedConfig, inputFieldName, inputObjectTypeName);
+  }
+
+  function mapDirective(
+    config: GraphQLDirectiveNormalizedConfig,
+  ): Maybe<GraphQLDirectiveNormalizedConfig> {
+    const mappedConfig = {
+      ...config,
+      args: mapArgs(config.args, config.name, undefined),
+    };
+    const mapper = configMapperMap[SchemaElementKind.DIRECTIVE];
+    return mapper == null ? mappedConfig : mapper(mappedConfig);
+  }
+}
+
+const stdTypeMap = new Map(
+  [...specifiedScalarTypes, ...introspectionTypes].map((type) => [
+    type.name,
+    type,
+  ]),
+);
```

Comment:
It seems this is now duplicated between https://github.com/graphql/graphql-js/blob/ef042cf131b27979269ce5b0e5ef1caac22b3d1d/src/utilities/extendSchema.ts#L574-L579 and https://github.com/graphql/graphql-js/blob/ef042cf131b27979269ce5b0e5ef1caac22b3d1d/src/utilities/mapSchemaConfig.ts#L459-L464
One of them should suffice :-)